cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/GcpMetadataHttpHandlerInternal.java

```diff
@@ -139,31 +157,37 @@ public void token(HttpRequest request, HttpResponder responder,
       // needed when initializing
       // io.cdap.cdap.common.guice.DFSLocationModule$LocationFactoryProvider#get
       // in io.cdap.cdap.internal.app.worker.TaskWorkerTwillRunnable.
+      LOG.warn("The GCP Metadata Task Context has been identified as null.");
       GcpTokenResponse gcpTokenResponse = new GcpTokenResponse("Bearer", "invalidToken", 3599);
       responder.sendJson(HttpResponseStatus.OK, GSON.toJson(gcpTokenResponse));
       return;
     }
 
     try {
       // fetch token from credential provider
-      GcpTokenResponse gcpTokenResponse =
-          Retries.callWithRetries(() -> fetchTokenFromCredentialProvider(scopes),
-              RetryStrategies.fromConfiguration(cConf, Constants.Service.TASK_WORKER + "."));
+      ProvisionedCredential provisionedCredential =
+          credentialLoadingCache.get(
+              new ProvisionedCredentialCacheKey(this.gcpMetadataTaskContext, scopes));
+      GcpTokenResponse gcpTokenResponse = new GcpTokenResponse("Bearer",
+          provisionedCredential.get(),
+          Duration.between(Instant.now(), provisionedCredential.getExpiration()).getSeconds());
       responder.sendJson(HttpResponseStatus.OK, GSON.toJson(gcpTokenResponse));
       return;
-    } catch (NotFoundException e) {
-      // if credential identity not found,
-      // fallback to gcp metadata server for backward compatibility.
-    } catch (Exception ex) {
-      if (ex instanceof HttpErrorStatusProvider) {
+    } catch (ExecutionException e) {
+      if (e.getCause() instanceof NotFoundException) {
+        // if credential identity not found,
+        // fallback to gcp metadata server for backward compatibility.
+      } else if (e.getCause() instanceof HttpErrorStatusProvider) {
         HttpResponseStatus status = HttpResponseStatus.valueOf(
-            ((HttpErrorStatusProvider) ex).getStatusCode());
-        responder.sendJson(status, exceptionToJson(ex));
+            ((HttpErrorStatusProvider) e.getCause()).getStatusCode());
+        responder.sendJson(status, exceptionToJson((Exception) e.getCause()));
+        return;
       } else {
```

Comment:
No need to have `else` after the `if` block returned.

Reply:
there are three blocks:
```
if {
// do something
} else if {
// do something
return;
} else {
// do something 
return;
}
```
If I don't add an `else`, it will return before than required.